controllers/replica_calculator.go

```diff
@@ -212,6 +214,89 @@ func (c *ReplicaCalculator) GetResourceReplicas(logger logr.Logger, target *auto
 	return ReplicaCalculation{replicaCount, utilizationQuantity, timestamp, int32(readyPodCount)}, nil
 }
 
+func Max(x, y int32) int32 {
```

Comment:
this function can be private and be rename to `maxInt32`

```suggestion
func maxInt32(x, y int32) int32 {
```

Reply:
I see lot of: 
`int32(math.Ceil(float64(..),float64(..))`, `int32(math.Mod(float64(..),float64(..))` or `int32(math.Floor(float64(..),float64(..))` in the following code change.

It would be nice to create a package `math32` to extract these functions and make the code more pleasant to read.

Reply:
done